.github/workflows/report-checks.yml

```diff
@@ -83,10 +85,36 @@ jobs:
               }
             }
           }
+          
+          async function deleteOldBotComments(pr_number) {
+            await github.paginate(
+                'GET /repos/{owner}/{repo}/pulls/{pull_number}/comments{?sort,direction,since,per_page,page}',
+                {
+                  owner: owner,
+                  repo: repo,
+                  pull_number: pr_number
+                },
+                (response => response.data.filter(
+                    comment => comment.user.login === 'orbitprofiler-bot').map(
+                    comment => comment.id))).then(
+                comment_ids => {
+                  comment_ids.forEach(comment_id => {
+                    console.log(`deleting comment: ${comment_id}`);
+                    github.request(
+                        'DELETE /repos/{owner}/{repo}/pulls/comments/{comment_id}', {
+                          owner: owner,
+                          repo: repo,
+                          comment_id: comment_id
+                    });
+                  });
+                }
+            );
+          }
+          
+          deleteOldBotComments(process.env.PR_NUMBER);
```

Comment:
I think we might have concurrency issues here, because the delection will run in parallel to adding the new comments. I would try to delete first and then add the new comments.

To do that we need an `await` here, but we also need to make sure that the returned promise completes not before all the deletions are done.

So this is how I would do it:

```js
const comments = await github.paginate(url, params);
const comment_ids = comments.filter(...).map(...);
await Promise.all(comment_ids.map(async (comment_id) => { await github.request(...); }));
});
```

Reply:
Agreed on better readability. Done.